semgrep-core/bin/Main.ml

```diff
@@ -380,8 +382,20 @@ let parse_generic lang file =
   Naming_AST.resolve lang ast;
   Constant_propagation.propagate lang ast;
   (*e: [[Main_semgrep_core.parse_generic()]] resolve names in the AST *)
-  ast
+  Left ast
```

Comment:
What does `Left` and `Right` do?

Reply:
Left and Right are constructors defined in pfff/.../Common.ml
They are a pretty standard idiom in OCaml and haskell to represent an alternative between 2 types.
It's defined as  type ('a, 'b) either = Left of 'a | Right of 'b